cocos/core/renderer/scene/directional-light.ts

```diff
@@ -50,40 +50,40 @@ export class DirectionalLight extends Light {
     }
 
     // in Lux(lx)
-    set illuminance (illum: number) {
+    get illuminance (): number {
         const isHDR = (legacyCC.director.root).pipeline.pipelineSceneData.isHDR;
         if (isHDR) {
-            this._illuminance = illum;
+            return this._illuminanceHDR;
         } else {
-            this._illuminanceLDR = illum;
-        }
-
-        if (JSB) {
-            (this._nativeObj as NativeDirectionalLight).setIlluminance(this._illuminance);
-            (this._nativeObj as NativeDirectionalLight).setIlluminanceLDR(this._illuminanceLDR);
+            return this._illuminanceLDR;
         }
     }
-
-    get illuminance (): number {
+    set illuminance (value: number) {
         const isHDR = (legacyCC.director.root).pipeline.pipelineSceneData.isHDR;
         if (isHDR) {
-            return this._illuminance;
+            this.illuminanceHDR = value;
         } else {
-            return this._illuminanceLDR;
+            this.illuminanceLDR = value;
         }
     }
 
-    set illuminanceHDR (illum: number) {
-        this._illuminance = illum;
+    get illuminanceHDR () {
+        return this._illuminanceHDR;
+    }
+    set illuminanceHDR (value: number) {
+        this._illuminanceHDR = value;
         if (JSB) {
-            (this._nativeObj as NativeDirectionalLight).setIlluminance(illum);
+            (this._nativeObj as NativeDirectionalLight).setIlluminanceHDR(value);
         }
     }
 
-    set illuminanceLDR (illum: number) {
-        this._illuminanceLDR = illum;
+    get illuminanceLDR () {
+        return this._illuminanceLDR;
+    }
+    set illuminanceLDR (value: number) {
```

Comment:
`illum` is not a word, it's preferred to use correct wording to improve readability